src/contracts/tokens/StakingManager.sol

```diff
@@ -0,0 +1,452 @@
+
+import {IStakingManager} from "@interfaces/tokens/IStakingManager.sol";
+
+import {Authorizable} from "@contracts/utils/Authorizable.sol";
+import {Modifiable} from "@contracts/utils/Modifiable.sol";
+
+/**
+ * @title  StakingManager
+ * @notice This contract is used to manage staking positions
+ *         and to distribute staking rewards
+ */
+contract StakingManager is Authorizable, Modifiable, IStakingManager {
+    // --- Registry ---
+
+    /// @inheritdoc IStakingManager
+    IProtocolToken public protocolToken;
+
+    /// @inheritdoc IStakingManager
+    IStakingToken public stakingToken;
+
+    // --- Params ---
+
+    /// @inheritdoc IStakingManager
+    // solhint-disable-next-line private-vars-leading-underscore
+    StakingManagerParams public _params;
+
+    /// @inheritdoc IStakingManager
+    function params()
+        external
+        view
+        returns (StakingManagerParams memory _params)
+    {
+        return _params;
+    }
+
+    // --- Data ---
+
+    /// @inheritdoc IStakingManager
+    uint256 public cooldownPeriod = 3 days;
+
+    /// @inheritdoc IStakingManager
+    uint256 public stakedSupply;
+
+    /// @inheritdoc IStakingManager
+    mapping(address => uint256) public stakedBalances;
+
+    /// @inheritdoc IStakingManager
+    // mapping(address => PendingWithdrawal[]) public pendingWithdrawals;
+    mapping(address => PendingWithdrawal) public pendingWithdrawals;
+
+    mapping(address => RewardType) public _rewardTypes;
+
+    /// @inheritdoc IStakingManager
+    function rewardTypes(
+        uint256 _id
+    ) external view returns (RewardType memory _rewardType) {
+        return _rewardTypes[_id];
+    }
+
+    /// @inheritdoc IStakingManager
+    uint256 public rewards;
+
+    // --- Init ---
+
+    /**
+     * @param  _protocolToken Address of the ProtocolToken contract
+     * @param  _stakingToken Address of the StakingToken contract
+     */
+    constructor(
+        address _protocolToken,
+        address _stakingToken
+    ) Authorizable(msg.sender) {
+        stakingToken = IStakingToken(_stakingToken);
+        protocolToken = IProtocolToken(_protocolToken);
+    }
+
+    // --- Methods ---
+
+    /// @inheritdoc IStakingManager
+    function stake(address _account, uint256 _wad) external {
+        if (_account == address(0)) revert StakingManager_StakeNullReceiver();
+        if (_wad == 0) revert StakingManager_StakeNullAmount();
+
+        stakedBalances[_account] += _wad;
+
+        // Mint stKITE
+        stakingToken.mint(_account, _wad);
+
+        // transfer KITE
+        protocolToken.safeTransferFrom(msg.sender, address(this), _wad);
+
+        // Call stake in the reward pools
+        uint256 rewardCount = rewards;
+        for (uint256 _i = 0; i < rewardCount; _i++) {
+            RewardType memory rewardType = _rewardTypes[_i];
+            if (rewardType.isActive) {
+                IRewardPool rewardPool = rewardType.rewardPool;
+                rewardPool.stake(_wad);
```

Comment:
stake 1 KITE
get 1 stKITE
and then you get 1 stake in many different pools.
So the amount of KITE staked != the number of stake combined in all the pools.  staked KITE will be much less if there are many pools. Wont that make accounting weird? Would it make sense to add that the user has to choose which pools they want to focus on and supply an array of the percentage staked to each. or you want everyone to get pro-rata part of every pool?

Reply:
@cupOJoseph is this comment still applicable considering our latest conversation re: Staking Manager? The number staked in the pools is just a numerical representation of how many tokens are staked in the StakingManager.

We want everyone to get pro-rata part of every pool. Convex does something similar where they use weighting and you can choose a ratio of which types of tokens you want rewards in (e.g. stable coins vs gov tokens). But choosing not to do that as it adds unneeded complexity.

Reply:
@cupOJoseph is this comment still applicable considering our latest conversation re: Staking Manager? The number staked in the pools is just a numerical representation of how many tokens are staked in the StakingManager.

We want everyone to get pro-rata part of every pool. Convex does something similar where they use weighting and you can choose a ratio of which types of tokens you want rewards in (e.g. stable coins vs gov tokens). But choosing not to do that as it adds unneeded complexity.

Reply:
Since its all relative it makes sense and this will work just fine. Everyone gets a pro-rata amount of weight in every pool based on their KITE staked.